src/Dialect/ONNX/ONNXOps/Additional/Custom.cpp

```diff
@@ -28,12 +28,40 @@ using namespace onnx_mlir;
 
 LogicalResult ONNXCustomOp::inferShapes(
```

Comment:
Is there any reason not to reuse the "ShapeHelper" infra? Since shape helpers are used, among others, for dynamic analysis, there would be great benefit in integrating it within that framework.

@tungld would you concur?

Reply:
@AlexandreEichenberger I put in a new version that supports ShapeHelper interface. 
Though the UnaryOpShapeHelper is used, the semantics is similar to onnx's propagating shape info. I used attribute value `SameAs` for multiple input. The reason for multiple inputs is that we can bind the symbol of multiple inputs for dim analysis (@tungld ). It may not be a good idea to make UnaryOpShapeHelper a subclass of BroadcastOpShapeHelper.
The special case for CustomOp is that it behaves differently based on the attribute value of this Op. In this implementation, I used multiple inheritance and call the corresponding subclass based on the attribute value. In order to do that, the UnaryOpShapeHelper and BroadcastOpShapeHelper has to be virtual and constructor has to initialize all the subclasses explicitly.